packages/ui-patterns/CommandMenu/prepackaged/DocsSearchLocal/DocsSearchLocal.shared.llm.ts

```diff
@@ -0,0 +1,27 @@
+import type { Tensor } from '@xenova/transformers'
+
+const MODEL = 'Supabase/gte-small'
+
+export interface Extractor {
+  extract: (input: string) => Promise<Tensor>
+}
+
+export function setupSingletonExtractor() {
```

Comment:
Our docs and most places will call this `embed()` vs `extract()` - maybe we should use similar terminology.

With respect to the singleton pattern - it might be clearer to just rename `setupSingletonExtractor()` itself to just `embed()` which still uses a singleton pattern but hides the details ([example](https://github.com/supabase-community/postgres-new/blob/358ad8b66db774b85fd8c1ee9a688cd623b58118/apps/postgres-new/lib/embed/index.ts#L6-L31)). Might be easier for new devs to understand when they see it for the first time.

Reply:
Also worth considering - inference is somewhat CPU intensive so it's usually better to offload to Web Worker so that the main thread/UI isn't blocked. You can use [`comlink`](https://github.com/GoogleChromeLabs/comlink) to simplify communication with the web worker.

Example of how we do this in `postgres-new`:
https://github.com/supabase-community/postgres-new/blob/main/apps/postgres-new/lib/embed/index.ts

---

**Edit:** I see now that you might only be calling this from a Web Worker already. If so this might not be relevant.